fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/impl/content/registry/util/QuiltDeferringQueues.java

```diff
@@ -0,0 +1,181 @@
+			QuiltDeferringQueues.BLOCK.deferEntry(block);
+		} else if (value instanceof BlockState state) {
+			QuiltDeferringQueues.BLOCK.deferEntry(state.getBlock());
+		} else if (value instanceof ReversibleBlockEntry entry) {
+			return QuiltDeferringQueues.BLOCK.deferEntry(entry.block());
+		} else if (value instanceof Item item) {
+			QuiltDeferringQueues.ITEM.deferEntry(item);
+		} else if (value instanceof GameEvent event) {
+			QuiltDeferringQueues.GAME_EVENT.deferEntry(event);
+		} else {
+			return false;
+		}
+
+		return true;
+	}
+
+	public static <V> boolean isEntryDeferred(V value) {
+		if (value instanceof Block block) {
+			return QuiltDeferringQueues.BLOCK.deferredEntries.contains(block);
+		} else if (value instanceof BlockState state) {
+			return QuiltDeferringQueues.BLOCK.deferredEntries.contains(state.getBlock());
+		} else if (value instanceof ReversibleBlockEntry entry) {
+			return QuiltDeferringQueues.BLOCK.deferredEntries.contains(entry.block());
+		} else if (value instanceof Item item) {
+			return QuiltDeferringQueues.ITEM.deferredEntries.contains(item);
+		} else if (value instanceof GameEvent event) {
+			return QuiltDeferringQueues.GAME_EVENT.deferredEntries.contains(event);
+		}
+
+		return false;
+	}
+
+	public static void updateOmniqueue() {
+		var entriesToRemove = new ArrayList<>();
+
+		for (var entry : OMNIQUEUE.entrySet()) {
+			var entriesToRemove2 = new ArrayList<>();
+
+			for (var listEntry : entry.getValue()) {
+				if (!isEntryDeferred(listEntry.entry()) && !isEntryDeferred(listEntry.value())) {
+					entry.getKey().put(listEntry.entry(), listEntry.value());
+					entriesToRemove2.add(listEntry.entry());
+				}
+			}
+
+			entry.getValue().removeAll(entriesToRemove2);
+			if (entry.getValue().size() == 0) entriesToRemove.add(entry.getKey());
+		}
+
+		entriesToRemove.forEach(entry -> OMNIQUEUE.remove(entry));
+	}
+
+	public static class DeferringQueue<K> {
+		public List<K> deferredEntries;
+		private boolean active;
+		private final Registry<K> registry;
+
+		public DeferringQueue(Registry<K> registry) {
+			this.deferredEntries = new ArrayList<>();
+			this.active = false;
+			this.registry = registry;
+		}
+
+		public boolean deferEntry(K entry) {
+			if (registry.getKey(entry).isEmpty()) {
+				this.deferredEntries.add(entry);
+				this.activate();
+
+				return true;
+			}
+
+			return false;
+		}
+
+		public void activate() {
+			if (!this.active) {
+				if (CRASH_ON_DEFERRING_ENTRY.toBooleanOrElse(QuiltLoader.isDevelopmentEnvironment())) {
+					if (CRASH_ON_DEFERRING_ENTRY == TriState.DEFAULT) {
+						LOGGER.error("An unregistered entry on the " + this.registry + " registry was attempted to be registered in a content registry through the Quilted Fabric API bridge! Due to this happening on a dev environment, the game will proceed to crash now. This crash may be disabled by disabling the \"quilted_fabric_api.quilted_fabric_content_registries_v0.crash_on_deferring_entry\" property.");
```

Comment:
For one thing this, and the other logger messages here, _need_ to be split into multiple lines for readability's sake.
And... how do you "disable" a property, exactly? This should be reworded to something like:
```
This crash may be disabled by setting the "quilted_fabric_api.quilted_fabric_content_registries_v0.crash_on_deferring_entry" property to "OFF".
```